tutorials/ab3TracingCode.md

```diff
@@ -281,31 +281,224 @@ Here are some quick questions you can try to answer based on your execution path
+* Reason: At least one field to edit must be provided.
+* Where the exception is thrown: `EditCommandParser#parse()`
+
+    ```java {.line-numbers highlight-lines="4"}
+    public EditCommand parse(String args) throws ParseException {
+        ...
+        if (!editPersonDescriptor.isAnyFieldEdited()) {
+            throw new ParseException(EditCommand.MESSAGE_NOT_EDITED);
+        }
+        ...
+    }
+    ```
+
+* Where the exception is handled: `MainWindow#executeCommand()`
+  </panel>
+
+<panel header="5. `edit 1 n/アリス ユー`" no-close>
+
+* Exception Thrown: ParseException
+* Reason: Names should only contain alphanumeric characters and spaces, and it should not be blank.
+* Where the exception is thrown: `ParserUtil#parseName()`
+
+    ```java {.line-numbers highlight-lines="4"}
+    public static Name parseName(String name) throws ParseException {
+        ...
+        if (!Name.isValidName(trimmedName)) {
+            throw new ParseException(Name.MESSAGE_CONSTRAINTS);
+        }
+        ...
+    }
+    ```
+
+* Where the exception is handled: `MainWindow#executeCommand()`
+  </panel>
+
+<panel header="6. `edit 1 t/one t/two t/three t/one`" no-close>
+
+* No exception thrown.
+* Reason: The command is correctly formatted and will edit the tags of the person at index 1 to "one", "two" and "three".
+
+    <panel header="Why didn’t the second tag with value “one” get added?">
+  
+    Duplicate values are handled by the `ParserUtil#parseTags()` method because tags are added to a HashSet. The HashSet class inherently handles duplicates by not allowing any equal elements to be added. Therefore, any duplicate tags are not added.<br>
+    Read more on the `add` method of the HashSet class <a href="https://docs.oracle.com/javase/8/docs/api/java/util/HashSet.html#add-E-">here</a>.
+    </panel>
+
+</panel>
+
+<br>
 
 2.  What components will you have to modify to perform the following
     enhancements to the application?
 
-    1.  Make command words case-insensitive
+<panel header="1. Make command words case-insensitive" no-close>
 
-    2.  Allow `delete` to remove more than one index at a time
+1. Modify `AddressBookParser#parseCommand()` to convert command words to lowercase before parsing.
 
-    3.  Save the address book in the CSV format instead
+    ```java {.line-numbers highlight-lines="3['.toLowerCase()']"}
+    public Command parseCommand(String userInput) throws ParseException {
+        ...
+        final String commandWord = matcher.group("commandWord").toLowerCase();
+        final String arguments = matcher.group("arguments");
+    }
+    ```
+</panel>
+
+<panel header="2. Allow `delete` to remove more than one index at a time" no-close>
 
-    4.  Add a new command
+1. Modify `DeleteCommandParser` to parse a list of indices.
+2. Update `DeleteCommand` to take in a list of indices.
+
+   <box type="info" seamless>
+  
+      Remember to update other usages of `DeleteCommand` class to handle the change in type of Index.
```

Comment:
Hi, really comprehensive answers here!
Just a nit: 'to handle the change in type of argument' might be better, I think.

Reply:
Ok, done!